### PR TITLE
linux_libsqlite.so_path_fixed

### PR DESCRIPTION
**Pull request by Mine-Ranger**
This fixes the error in linux as the path found it automatically was wrong so I made a check for linux right above mac os one

### CODE REVIEW

File: lua/sqlite/defs.lua [hunk 1]
```diff
@@ -19,8 +19,11 @@ local clib = (function()
   end
 
   local clib_path = path
-    or (function() --- try to find libsqlite. Macos support only.
+    or (function() --- try to find libsqlite.Linux and Macos support only.
       local os = luv.os_uname()
+      if os.sysname == "Linux" then
```
Comment (kkharji) [inline lua/sqlite/defs.lua:24]: What if the architecture is different? e.g. arm. Is this is unified across all linux distros?

Comment (Mine-Ranger) [reply lua/sqlite/defs.lua:30]: > What if the architecture is different? e.g. arm. Is this is unified across all linux distros?

Thank you for your input.
I have now implemented a path checker in the code and it now works on (Ubuntu,Fedora,Arch) tested in a docker container
I am not completely sure about arm64 (as I do not have one) but I assume it should work there as well as I tried to cover all possible path for libsqlite3.so

Comment (kkharji) [reply lua/sqlite/defs.lua:30]: That sufficient, users can always set a variable to where the clib is. 

Thanks @Mine-Ranger

**Review by kkharji — CHANGES_REQUESTED** [review]
The CI failing because of lint issue. After fixing that it should be mergeable

**Review by kkharji — APPROVED** [review]
Tests seems to be broken for some reason. I will follow up this pr with a fix.

Thanks @Mine-Ranger

**Pull request merged**